src/jobs/get-br-dependencies.yml

```diff
@@ -0,0 +1,76 @@
+parameters:
+  exec:
+    type: executor
+  aws-access-key-id:
+    type: env_var_name
+    default: AWS_ACCESS_KEY_ID
+  aws-secret-access-key:
+    type: env_var_name
+    default: AWS_SECRET_ACCESS_KEY
+  aws-region:
+    type: env_var_name
+    default: AWS_REGION
+  cache-bucket-uri:
+    type: env_var_name
+    default: CACHE_BUCKET_URI
+  resource-class:
+    type: string
+    default: medium
+  hex-validate:
+    type: boolean
+    default: true
+  env-setup:
+    type: steps
+    default: []
+  save-dl-cache:
+    type: boolean
+    default: true
+  save-public-cache:
+    type: boolean
+    default: false
+executor: << parameters.exec >>
+resource_class: << parameters.resource-class >>
+steps:
+  - checkout
+  - steps: << parameters.env-setup >>
+  - install-elixir:
+      version: $ELIXIR_VERSION
+  - install-hex-rebar
+  - install-nerves-bootstrap
+  - run:
+      name: Install AWS CLI
+      command: |
+        apt install -y python3-pip
+        pip install awscli
+
+        aws configure set aws_access_key_id ${<< parameters.aws-access-key-id >>}
+        aws configure set aws_secret_access_key ${<< parameters.aws-secret-access-key >>}
+        aws configure set region ${<< parameters.aws-region >>}
+  - mix-deps-get
+  - restore-nerves-downloads
+  - when:
+      condition: << parameters.hex-validate >>
+      steps:
+        - run:
+            name: Validate Hex package
+            command: mix hex.build
+  - run:
+      name: Get Buildroot dependencies
+      command: |
+        ./deps/nerves_system_br/create-build.sh nerves_defconfig ~/temp_nerves_system
+        cd ~/temp_nerves_system
+        make source
+  - when:
+      condition: << parameters.save-dl-cache >>
+      steps:
+      - save_cache:
+          name: Save Nerves downloads
+          key: nerves-dl-{{ .Branch }}-{{ .Revision }}
```

Comment:
I'm wondering if we need to adapt this cache key. It works well with multiple builds of the same branch/PR, but when we merge it and main rebuilds (or a release rebuilds), we'll miss out on using our new cache

Reply:
Although optimized for iterating on the same branch/PR, it should work "good enough" when merging to main due to these lines being in the restore task. This will at least grab the last cache for main, or punt and grab any cache it can.

```yaml
- nerves-dl-main-
- nerves-dl-
```

However, if you want to optimize better for the PR merge scenario, what do you think about basing the cache on the hash of `nerves_defconfig`?

```yaml
nerves-dl-{{ checksum "nerves_defconfig" }}
```

Reply:
What's the downside of not caring and always restoring? Worst case you'll get a tarball that you don't need.

Reply:
We talked a bit offline and the TL;DR is that this is just a CircleCI-ism, so we'll go with it for now